src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -90,48 +75,47 @@ internal AuthenticationOptions BaseOptions
             }
         }
 
-        private static void OnSendingHeaderCallback(object state)
+        private async Task OnResponseStartingCallback()
```

Comment:
This was supposed to stay static so we only allocate one delegate ever, not one per request.